portfolio/src/main/webapp/commentsection.js

```diff
@@ -13,27 +13,69 @@
 // limitations under the License.
 
 function getComments() {
-    const showNum = getCommentLimit();
-    document.getElementById('shownum').value = parseInt(showNum);
-    fetch('/comment-data?shownum='+showNum).then(response => response.json()).then((comments) => {
+    const perBlock = getCommentsPerBlock();
+    document.getElementById('perblock').value = parseInt(perBlock);
+    fetch('/comment-data?perblock='+(perBlock*blockIdx)).then(response => response.json()).then((data) => {
+        let comments = data.comments;
         const commentsContainer = document.getElementById("commentsection-container");
         commentsContainer.innerHTML = "";
         comments.forEach((line) => {
             commentsContainer.appendChild(createComment(line));
         });
+        if(data.totalCommentCount <= perBlock * blockIdx) {
+            let showmorebutton = document.getElementById("showmorebutton");
+            showmorebutton.style.display = "none";
+        }
     });
 }
 
+var blockIdx = 1;
```

Comment:
I think it's fine as is because this value gets reset whenever the page is refreshed. If we delete all the comments, we won't be able to see more comments until we refresh the page anyways, so this wouldn't make a difference.

Reply:
Ah, make sense.